server/db/actions/Dog.js

```diff
@@ -0,0 +1,61 @@
+import dbConnect from "../dbConnect";
+import Dog from "../models/Dog";
+import Log from "../models/Log";
+import User from "../models/User";
+
+/**
+ * Updates an existing dog in the database
+ * @param {*} dogId ID of the Dog Object to be updated
+ * @param {*} dogData Dog Object that has been parsed by Zod for validity
+ * @returns the Dog Object prior to the update; error otherwise
+ */
+export async function updateDog(dogId, dogData) {
+  await dbConnect();
+
+  // recentLogs array
+  if (dogData.recentLogs && dogData.recentLogs.length) {
+    for (let i = 0; i < dogData.recentLogs.length; i++) {
+      if (!(await Log.findById(dogData.recentLogs[i]))) {
+        throw new Error("Log ID is not present in database");
```

Comment:
Theses error messages do not propagate because the `.catch()` has a static message `"Unable to update dog"`

Reply:
Updated the catch statement in [id].js
I was confusing the valid Object ID check and Object ID exists in the DB check 😅